### ISSUE TITLE
feat: ux: set max number of columns in repr(Table)

### ISSUE THREAD

**Issue by NickCrews** [ibis-project/ibis#5477]
### Is your feature request related to a problem?

When I'm quickly working in a notebook, I often look at a table. But it cuts off the number of columns I see, which I find annoying. I want to be able to see all the columns. in pandas I could do `pandas.options.display.max_columns = int | None`
I would like to be able to do the same thing.

Note that there is also a separate bug that would need to be addressed, where the ibis code that determines how many columns to show uses a while loop that is order O(n), so for very large widths it is slow.

### Describe the solution you'd like

 Currently I monkeypatch in my own method as:

```python
def set_table_repr_width(width: int = 1000):
    """Set the width (in chars) that tables are repr'd at in Jupyter notebooks.

    Often useful to call at the beginning of a notebook.
    """

    # You can update the width later if you want
    ibis.options.repr.max_columns = width

    def my__rich_console__(self, console, options):
        from ibis.expr.types.pretty import to_rich_table
        from rich.text import Text

        if not ibis.options.interactive:
            return console.render(Text(self._repr()), options=options)

        width = ibis.options.repr.max_columns
        table = to_rich_table(self, width)
        options.max_width = width
        return console.render(table, options=options)

    Table.__rich_console__ = my__rich_console__
```

### What version of ibis are you running?

target 5.0?

### What backend(s) are you using, if any?

all

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jcrist** [comment]
> But it cuts off the number of columns I see, which I find annoying.

How would this work for a large number of columns on a narrow terminal? Right now we only display as much information as your window has room to display. What kind of display are you hoping to see?

> Note that there is also a separate bug that would need to be addressed, where the ibis code that determines how many columns to show uses a while loop that is order O(n), so for very large widths it is slow.

Do you have a reproducible example that shows the performance issue here?

**Comment by NickCrews** [comment]
> Do you have a reproducible example that shows the performance issue here?

Use the above monkeypatch as `set_table_repr_width(width: int = 1_000_000)` and then repr a wide table. Sorry, nothing totally scriptable.

> What kind of display are you hoping to see?

In a .ipynb file in vscode, it just works:

<img width="631" alt="image" src="https://user-images.githubusercontent.com/10820686/217386165-e21dd750-07c4-4d48-a2ba-fdb13cc96546.png">

I can scroll to see the whole frame. Note the scrollbar on the bottom. I'm not sure how terminals really work and how this would actually behave. I think the best you could hope for on a "dumb" terminal is for the long lines to just get wrapped, and it would look bad. This is all pandas does.

At the least, setting `max_columns = None` should just show all columns, and its up to the user to have their terminal configured correctly.

**Comment by jcrist** [comment]
This should be fixed by #5494. Based on what you're doing here, I think you'd want to set:

```python
ibis.options.repr.interactive.max_columns = None
```

to always display the full table width in the notebook. I believe this has also addressed the performance issue.

**Comment by NickCrews** [comment]
Thanks @jcrist, that looks awesome!
